### PR TITLE
Introduction to Topology2.0

### PR DESCRIPTION
**Pull request by ranj063**
Alsa-lib changes: https://github.com/ranj063/alsa-lib/tree/topic/topology2-prep-wip
Alsa-utils changes:https://github.com/ranj063/alsa-utils/commits/topic/topology2.0

### CODE REVIEW

File: tools/topology2/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+# Array of "input-file-name;output-file-name;"
+set(TPLGS
+	"sof-cnl-nocodec\;sof-cnl-nocodec\;"
```
Comment (marc-hb) [inline tools/topology2/CMakeLists.txt:5]: As mentioned on the chat, you can't have `${input.conf}` == `${output.conf}`, no build system will support that.

**Comment by marc-hb** [comment]
There were a number of CI issues recently (#3992 and others). @ranj063 please `git commit --amend` and force-push the same code if you want to re-trigger everything.

**Comment by gkbldcig** [comment]
Can one of the admins verify this patch?

**Comment by lgirdwood** [comment]
For v1.9 - a reminder to merge after ALSA changes merged.

**Comment by marc-hb** [comment]
This PR should land the new `.tplg` files in the `sof/tools/build_tools/topology2/` build directory. M4 topology files currently land in `sof/tools/build_tools/topology/`.

Here's a proposition to implement a gradual transition for topology v1 to v2 once this PR is merged.

- Topology v1 keeps building as usual, no change
- After building both, `build-tools.sh` copies the entire `sof/tools/build_tools/topology/` to `sof/tools/build_tools/topology-v1/` as a backup and for anyone or anything that wants to compare v1 and v2
- Then a gradually growing subset of `topology2/*.tplg` files overwrites the `topology/*.tplg` files of the same name. This can be done with simple, version-controlled `rsync --ignore-times --include-from=active_v2_topologies.txt` 
- Users start using v2 files without having to change a single line of code.

git churn is minimized.

**Comment by lgirdwood** [comment]
@marc-hb I assume you will do an update to ```build_tools.sh```

**Comment by marc-hb** [comment]
No I was just answering an off-line question, sorry for the confusion.

**Comment by ranj063** [comment]
closing this for now. Will post a new PR with smaller commits

**Pull request closed without merging**